src/common/libutil/read_all.h

```diff
@@ -13,7 +13,15 @@
 
 #include <sys/types.h>
 
+#include "src/common/libutil/sha1.h"
+
+/* sha1.h defines SHA1_DIGEST_SIZE */
+#define SHA1_DIGEST_SIZE_STR (SHA1_DIGEST_SIZE*2 + 1)
+
 ssize_t write_all (int fd, const void *buf, size_t len);
 ssize_t read_all (int fd, void **bufp);
+/* file_digest returns length of data read from fd */
+ssize_t file_digest (int fd, uint8_t digest[SHA1_DIGEST_SIZE]);
+ssize_t file_digest_str (int fd, char digeststr[SHA1_DIGEST_SIZE_STR]);
```

Comment:
Hmm, if you call `read_all()` on an open file and then `file_digest_str()` you won't get what you think because nothing resets the file pointer.

I think the helper functions here could provide simpler interfaces not involving file descriptors, and that would simplify the callers.  Maybe something like:
```c
char *digest_file (const char *pathname);
char *digest_buf (const void *buf, size_t len);
```
and then maybe they belong in their own source file?

Reply:
yeah, i was trying not to introduce new files to `libutil`, but perhaps i shouldn't have tried too hard to avoid it.